File: 01_Language/01_Functions/go/src/functions/gethostbyaddr.go
package functions

import (
	"log"
	"net"
)

func Gethostbyaddr(ip string) string {
	names, err := net.LookupAddr(ip)
	if err != nil {
		log.Println(err)
		return ip
	}
	return names[0]
}

File: 01_Language/01_Functions/java/src/main/java/com/onevgo/functions/Gethostbyaddr.java
package com.onevgo.functions;

import java.net.InetAddress;
import java.net.UnknownHostException;

public class Gethostbyaddr {
    public static String gethostbyaddr(String addr) {
        try {
            return InetAddress.getByName(addr).getHostName();
        } catch (UnknownHostException e) {
            e.printStackTrace();
        }
        return addr;
    }

    public static void main(String[] args) {
        System.out.println(gethostbyaddr("193.112.23.137"));
    }
}
